analyses/cell-type-ewings/cell_type_annotations/SCPCS000490/SCPCL000822_tumor-cell-validation.Rmd

```diff
@@ -0,0 +1,492 @@
+  facet_wrap(vars(method))
+```
+
+This plot shows us that there is a pretty clear consensus that most of the cells in the bottom left group are tumor cells. 
+From the original marker gene analysis we know that not all marker genes are expressed and that there is quite a bit of variance in expression. 
+Because of that and the fact that both CNV inference methods show similar annotations, we will classify any cells that are labeled as tumor by both `CopyKAT` and `InferCNV` as tumor cells. 
+
+```{r}
+# create a combined classification between inferCNV and copyKAT
+# cells that are classified as tumor by both are kept as tumor, all other are normal 
+coldata_df <- coldata_df |> 
+  dplyr::mutate(
+    combined_cnv_classification = dplyr::if_else(
+      cnv_proportion_classification == "Tumor" & copykat_classification == "Tumor",
+      "Tumor",
+      "Normal"
+    ),
+    # there are cells classified as NA by copykat, we want those to be normal 
+    combined_cnv_classification = dplyr::if_else(is.na(copykat_classification) & is.na(combined_cnv_classification), "Normal", combined_cnv_classification)
+  )
+```
+
+```{r}
+# look at which cells are tumor/normal
+# be sure to remove any duplicates 
+coldata_df |> 
+  dplyr::select(UMAP1, UMAP2, barcodes, combined_cnv_classification) |>
+  unique() |> 
+  ggplot(aes(x = UMAP1, y = UMAP2, color = combined_cnv_classification)) +
+  geom_point(alpha = 0.5, size = 0.5)
+```
+
+The remainder of the notebook will focus on validating that these in fact are tumor cells and all other cells are classified as normal. 
+
+## Validate tumor cells 
+
+### Marker gene expression 
+
+First, we will look at the expression of the tumor marker genes. 
+We expect that these should be expressed higher in the tumor cells than in the normal cells. 
+
+In this first plot, we show a density plot of marker gene expression coloring by the cell classification (tumor or normal). 
+For this plot, we will include all classifications, including the new combined CNV classification. 
+
+```{r}
+# calculate sum of the scaled gene expression values for plotting 
+marker_density_df <- coldata_df |> 
+  dplyr::select(barcodes, gene_symbol, gene_expression, ends_with("classification")) |> 
+  unique() |> 
+  dplyr::group_by(gene_symbol) |>
+  # get z-scores for each gene
+  dplyr::mutate(transformed_gene_expression = scale(gene_expression)[, 1]) |> 
+  dplyr::ungroup() |> 
+  dplyr::group_by(barcodes) |> 
+  # add all the scores together 
+  dplyr::mutate(sum_exp = sum(transformed_gene_expression)) |>
+  # expand classifications column for easy faceting 
+  dplyr::ungroup() |> 
+  tidyr::pivot_longer(
+    cols = ends_with("classification"),
+    names_to = "method",
+    values_to = "classification"
+  )
+
+ggplot(marker_density_df, aes(x = sum_exp, color = classification)) +
+  geom_density() +
+  facet_wrap(vars(method))
+```
+
+This plot shows us that overall the tumor cells in each method have higher marker gene expression. 
```

Comment:
Probably worth saying that the `marker_gene_classification` plot here is circular, in that it just reflects the cutoff that was used for classification